src/legacy/core_plugins/embeddable_api/public/actions/apply_filter_action.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { Container, ContainerInput } from '../containers';
+import { IEmbeddable } from '../embeddables';
+import { APPLY_FILTER_TRIGGER, triggerRegistry } from '../triggers';
+import { Filter } from '../types';
+import { Action, ActionContext } from './action';
+import { actionRegistry } from '../actions';
+import { IncompatibleActionError } from './incompatible_action_error';
+import { IContainer } from '../containers/i_container';
+import { attachAction } from '../triggers/attach_action';
+
+interface ApplyFilterContainerInput extends ContainerInput {
+  filters: Filter[];
+}
+
+const APPLY_FILTER_ACTION = 'APPLY_FILTER_ACTION';
```

Comment:
++ though I'm going to keep using a string type for the id.  This will be used in phase 2 (e.g you will be able to have two actions of the same type, but with different configuration parameters attached to a single trigger). We can modify as needed when phase 2 actually gets underway.

Reply:
actually, this is going to be a pain to switch over to because it's going to mess all the functional tests up that rely on this being a string.  I would prefer to punt on this for now to make progress, but have no objection to a follow up that switches to symbols.